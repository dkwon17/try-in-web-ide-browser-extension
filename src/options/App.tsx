/*-----------------------------------------------------------------------------------------------
 *  Copyright (c) Red Hat, Inc. All rights reserved.
 *  Licensed under the MIT License. See LICENSE file in the project root for license information.
 *-----------------------------------------------------------------------------------------------*/

import { useEffect, useState } from "react";

import "@patternfly/react-core/dist/styles/base.css";
import "@patternfly/patternfly/utilities/Spacing/spacing.css";
import "@patternfly/patternfly/utilities/Float/float.css";

import {
    Button,
    Card,
    CardBody,
    Divider,
    Split,
    SplitItem,
    Form,
    FormGroup,
    TextInput,
    CardTitle,
} from "@patternfly/react-core";
import { ExclamationCircleIcon } from "@patternfly/react-icons";
import {
    Endpoint,
    getEndpoints,
    saveEndpoints,
} from "../preferences/preferences";
import { EndpointsList } from "./EndpointsList";
import "./styles/App.css";

export const App = () => {
    const [endpoints, setEndpoints] = useState<Endpoint[]>([]);
    const [newEndpointUrl, setNewEndpointUrl] = useState<string>("");
    
    type validate = "success" | "error" | "default";
    const [newEndpointStatus, setNewEndpointStatus] = useState<validate>("default");

    useEffect(() => {
        const fetchData = async () => {
            const endpoints = await getEndpoints();
            setEndpoints(endpoints);
        };
        fetchData().catch(console.error);
    }, []);

    const handleNewEndpointUrlChange = (newUrl: string, _event: React.FormEvent<HTMLInputElement>) => {
        setNewEndpointUrl(newUrl);
        if (newUrl === "") {
          setNewEndpointStatus("default");
        } else if (isUrl(newUrl)) {
          setNewEndpointStatus("success");
        } else {
          setNewEndpointStatus("error");
        }
    };

    const isUrl = (str: string) => {
        try {
            new URL(str);
        } catch {
            return false;
        }
        return true;
    }

    const addNewEndpoint = async () => {
        const sanitizedEndpoint = sanitizeEndpoint(newEndpointUrl);
        const newEndpoints = endpoints.concat({
            url: sanitizedEndpoint,
            active: false,
            readonly: false,
        });
        await saveEndpoints(newEndpoints);
        await setEndpoints(newEndpoints);
        setNewEndpointUrl("");
        setNewEndpointStatus("default");
    };

    const sanitizeEndpoint = (str: string) => {
        let res = str;
        while (res.charAt(res.length - 1) === "/") {
            res = res.substring(0, res.length - 1);
        }
        return res;
    }

    const setDefault = async (endpoint: Endpoint) => {
        const newEndpoints = [...endpoints];
        newEndpoints.forEach((e) => {
            e.active = e == endpoint;
        });
        await saveEndpoints(newEndpoints);
        await setEndpoints(newEndpoints);
    };

    const deleteEndpoint = async (endpoint: Endpoint) => {
        const newEndpoints = endpoints.filter((e) => e != endpoint);
        await saveEndpoints(newEndpoints);
        await setEndpoints(newEndpoints);
    };

    const list = endpoints.length && (
        <EndpointsList
            endpoints={endpoints}
            onClickSetDefault={setDefault}
            onClickDelete={deleteEndpoint}
        />
    );

    return (
        <Card>
            <CardTitle>Dev Spaces endpoints</CardTitle>
            <CardBody>
                {list}
                <Divider className="pf-u-mt-md pf-u-mb-md" />
                <Split>
                    <SplitItem className="form-text-input">
                        <Form>
                            <FormGroup
                                validated={newEndpointStatus}
                                helperTextInvalid="The new endpoint must be a URL"
                                helperTextInvalidIcon={<ExclamationCircleIcon />}>
                                <TextInput
                                    type="text"
                                    aria-label="new endpoint"
                                    validated={newEndpointStatus}
                                    value={newEndpointUrl}
                                    placeholder="Add endpoint"
                                    onChange={handleNewEndpointUrlChange}
                                />
                            </FormGroup>
                        </Form>
                    </SplitItem>
                    <SplitItem className="form-fill" isFilled></SplitItem>
                    <SplitItem>
                        <Button variant="primary" onClick={addNewEndpoint} isDisabled={newEndpointUrl.length === 0 || newEndpointStatus === "error"}>
                            Add
                        </Button>
                    </SplitItem>
                </Split>
            </CardBody>
        </Card>
    );
};
