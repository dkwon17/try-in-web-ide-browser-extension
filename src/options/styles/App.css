.pf-c-card {
    --pf-c-card__title--FontWeight: var(--pf-global--FontWeight--normal) !important;
    --pf-c-card__title--not--last-child--PaddingBottom: var(--pf-global--spacer--sm) !important;
    --pf-c-card__title--FontSize: var(--pf-global--FontSize--3xl) !important;
    --pf-c-card--first-child--PaddingTop: var(--pf-global--spacer--md) !important;
    --pf-c-card--child--PaddingBottom: var(--pf-global--spacer--xl) !important;
    --pf-c-card--child--PaddingRight: var(--pf-global--spacer--xl) !important;
    --pf-c-card--child--PaddingLeft: var(--pf-global--spacer--xl) !important;
    height: 100%;
}

/* styling the list of hostnames */
.pf-c-simple-list {
    --pf-c-simple-list__item-link--PaddingLeft: 0 !important;

    /* disable hover background change */
    --pf-c-simple-list__item-link--hover--BackgroundColor: var(--pf-global--BackgroundColor--light-100) !important;
    --pf-c-simple-list__item-link--Color: var(--pf-global--Color--100) !important;

    /* disable styles when list item is clicked on */
    --pf-c-simple-list__item-link--m-current--BackgroundColor: var(--pf-global--BackgroundColor--light-100) !important;
    --pf-c-simple-list__item-link--m-current--Color: var(--pf-global--Color--100) !important;
    --pf-c-simple-list__item-link--active--BackgroundColor: var(--pf-global--BackgroundColor--light-100) !important;
    --pf-c-simple-list__item-link--active--Color: var(--pf-global--Color--100) !important;
    --pf-c-simple-list__item-link--FontSize: var(--pf-global--FontSize--md) !important;
}

/* styling the dropdown menu button */
.pf-c-dropdown {
    --pf-c-dropdown__toggle--PaddingTop: 0 !important;
    --pf-c-dropdown__toggle--PaddingRight: 0 !important;
    --pf-c-dropdown__toggle--PaddingBottom: 0 !important;
    --pf-c-dropdown__toggle--PaddingLeft: 0 !important;
    --pf-c-dropdown__toggle--before--BorderWidth: 0 !important;
    --pf-c-dropdown--m-expanded__toggle--before--BorderBottomWidth: 0 !important;
    --pf-c-dropdown__toggle--focus--before--BorderBottomWidth: 0 !important;
    --pf-c-dropdown__toggle--active--before--BorderBottomWidth: 0 !important;
    --pf-c-dropdown__toggle--Color: var(--pf-global--palette--black-600) !important;
}

.pf-c-simple-list__item-link {
    cursor: default;
}

.form-text-input {
    width: 100%;
}

.form-fill {
    width: var(--pf-global--spacer--sm);
}

