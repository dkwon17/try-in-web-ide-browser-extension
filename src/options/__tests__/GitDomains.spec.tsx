/*-----------------------------------------------------------------------------------------------
 *  Copyright (c) Red Hat, Inc. All rights reserved.
 *  Licensed under the MIT License. See LICENSE file in the project root for license information.
 *-----------------------------------------------------------------------------------------------*/

import { fireEvent, render, screen, within } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import "@testing-library/jest-dom";

import { App } from "../App";
import { chrome } from "jest-chrome";

const preferencesMock = require("../../preferences/preferences");

jest.mock("../../preferences/preferences");

beforeEach(() => {
    chrome.permissions.contains.mockImplementation(async () => {
        return true;
    });
});

afterEach(() => {
    jest.clearAllMocks();
});

it("renders correctly with one domain", async () => {
    preferencesMock.setGitDomains(["https://github.example.com"]);
    const { findByText, asFragment } = render(<App />);
    await findByText("https://github.example.com");
    expect(asFragment()).toMatchSnapshot();
});

it("renders correctly with multiple domains", async () => {
    preferencesMock.setGitDomains([
        "https://github.example-1.com",
        "https://github.example-2.com",
        "https://github.example-3.com",
    ]);
    const { findByText, asFragment } = render(<App />);
    await findByText("https://github.example-1.com");
    await findByText("https://github.example-2.com");
    await findByText("https://github.example-3.com");
    expect(asFragment()).toMatchSnapshot();
});

it("should have the same number of endpoints as links in the endpoints list (1)", async () => {
    preferencesMock.setGitDomains(["https://github.example-1.com"]);
    const { findByText, findByTestId } = render(<App />);
    await findByText("https://github.example-1.com");

    const list = await findByTestId("git-domains-list");
    expect(list.querySelectorAll("li").length).toEqual(1);
});

it("should have the same number of endpoints as links in the endpoints list (2)", async () => {
    preferencesMock.setGitDomains([
        "https://github.example-1.com",
        "https://github.example-2.com",
        "https://github.example-3.com",
    ]);
    const { findByText, findByTestId } = render(<App />);
    await findByText("https://github.example-1.com");
    await findByText("https://github.example-2.com");
    await findByText("https://github.example-3.com");

    const list = await findByTestId("git-domains-list");
    expect(list.querySelectorAll("li").length).toEqual(3);
});

it("should enable 'Add' button when a valid URL is entered in the input box", async () => {
    preferencesMock.setGitDomains(["https://github.example.com"]);

    const { findByPlaceholderText } = render(<App />);
    const inputBox = await findByPlaceholderText(
        "Add GitHub Enterprise domain"
    );
    const domainsTab = await screen.findByTestId("git-domains-tab");
    const addButton = (await within(domainsTab).findByText("Add")).closest(
        "button"
    );

    fireEvent.change(inputBox, {
        target: { value: "https://github.example.com" },
    });
    expect(addButton.classList.contains("pf-m-disabled")).toBe(false);
});

it("should disable 'Add' button when invalid input is entered in the input box", async () => {
    const { findByPlaceholderText } = render(<App />);
    const inputBox = await findByPlaceholderText(
        "Add GitHub Enterprise domain"
    );
    const endpointsTab = await screen.findByTestId("git-domains-tab");
    const addButton = (await within(endpointsTab).findByText("Add")).closest(
        "button"
    );

    fireEvent.change(inputBox, { target: { value: "    " } });
    expect(addButton.classList.contains("pf-m-disabled")).toBe(true);

    fireEvent.change(inputBox, { target: { value: "invalidtext" } });
    expect(addButton.classList.contains("pf-m-disabled")).toBe(true);

    fireEvent.change(inputBox, { target: { value: "123412341234" } });
    expect(addButton.classList.contains("pf-m-disabled")).toBe(true);
});

it("should display error message when invalid input is entered in the input box", async () => {
    const { findByPlaceholderText, findByText } = render(<App />);
    const inputBox = await findByPlaceholderText(
        "Add GitHub Enterprise domain"
    );
    fireEvent.change(inputBox, { target: { value: "invalidtext" } });
    await findByText(
        "Provide the URL of your GitHub Enterprise instance, e.g.,\nhttps://github.example.com",
        { collapseWhitespace: false }
    );
});

it("should display error message when invalid input with invalid protocol is entered in the input box", async () => {
    const { findByPlaceholderText, findByText } = render(<App />);
    const inputBox = await findByPlaceholderText(
        "Add GitHub Enterprise domain"
    );
    fireEvent.change(inputBox, {
        target: { value: "test://github.example.com" },
    });
    await findByText(
        "Provide the URL of your GitHub Enterprise instance, e.g.,\nhttps://github.example.com",
        { collapseWhitespace: false }
    );
});

it("should display alert when host permissions already granted", async () => {
    const { findByPlaceholderText, findByText } = render(<App />);
    const inputBox = await findByPlaceholderText(
        "Add GitHub Enterprise domain"
    );
    const domainsTab = await screen.findByTestId("git-domains-tab");
    const addButton = (await within(domainsTab).findByText("Add")).closest(
        "button"
    );

    fireEvent.change(inputBox, {
        target: { value: "https://github.example.com" },
    });

    await act(() => {
        addButton.click();
    });

    await findByText(
        "Host permissions for https://github.example.com already granted"
    );
});

it("should remove host permission when git domain is removed", async () => {
    preferencesMock.setGitDomains(["https://github.example.com"]);

    const { findByText, findByLabelText } = render(<App />);
    const deleteButton = await findByLabelText(
        "Remove GitHub Enterprise domain https://github.example.com"
    );
    await act(() => {
        deleteButton.click();
    });

    const confirmRemoveBtn = await findByText("Remove");
    await act(() => {
        confirmRemoveBtn.click();
    });

    expect(chrome.permissions.remove).toHaveBeenCalledTimes(1);
});
