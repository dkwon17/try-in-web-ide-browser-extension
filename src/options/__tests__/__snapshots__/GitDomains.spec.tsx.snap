// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with multiple domains 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card"
    data-ouia-component-id="OUIA-Generated-Card-2"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <section
      class="pf-c-page__main-tabs pf-m-light pf-m-limit-width"
    >
      <div
        class="pf-c-page__main-body"
      >
        <div
          class="pf-c-tabs pf-m-page-insets"
          data-ouia-component-id="OUIA-Generated-Tabs-2"
          data-ouia-component-type="PF4/Tabs"
          data-ouia-safe="true"
          id="open-tabs-example-tabs-list"
        >
          <ul
            class="pf-c-tabs__list"
            role="tablist"
          >
            <li
              class="pf-c-tabs__item pf-m-current"
              role="presentation"
            >
              <button
                aria-controls="tabContent0"
                aria-selected="true"
                class="pf-c-tabs__link"
                data-ouia-component-type="PF4/TabButton"
                data-ouia-safe="true"
                id="pf-tab-0-open-tabs-example-tabs-list"
                role="tab"
                type="button"
              >
                <span
                  class="pf-c-tabs__item-text"
                >
                  Dev Spaces endpoints
                </span>
              </button>
            </li>
            <li
              class="pf-c-tabs__item"
              role="presentation"
            >
              <button
                aria-controls="tabContent1"
                aria-selected="false"
                class="pf-c-tabs__link"
                data-ouia-component-type="PF4/TabButton"
                data-ouia-safe="true"
                id="pf-tab-1-open-tabs-example-tabs-list"
                role="tab"
                type="button"
              >
                <span
                  class="pf-c-tabs__item-text"
                >
                  GitHub Enterprise domains
                </span>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-light pf-m-limit-width"
    >
      <div
        class="pf-c-page__main-body"
      >
        <section
          aria-labelledby="tabContent0"
          class="pf-c-tab-content"
          data-ouia-component-type="PF4/TabContent"
          data-ouia-safe="true"
          data-testid="dev-spaces-endpoints-tab"
          id="tabContent0"
          role="tabpanel"
          tabindex="0"
        >
          <div
            class="pf-c-tab-content__body"
          >
            No endpoints added yet
            <hr
              class="pf-c-divider pf-u-mt-md pf-u-mb-md"
            />
            <div
              class="pf-l-split"
            >
              <div
                class="pf-l-split__item form-text-input"
              >
                <form
                  class="pf-c-form"
                  novalidate=""
                >
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid="false"
                        aria-label="new endpoint"
                        class="pf-c-form-control"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-3"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe="true"
                        placeholder="Add endpoint"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </form>
              </div>
              <div
                class="pf-l-split__item pf-m-fill form-fill"
              />
              <div
                class="pf-l-split__item"
              >
                <button
                  aria-disabled="true"
                  aria-label="add endpoint"
                  class="pf-c-button pf-m-primary pf-m-disabled"
                  data-ouia-component-id="OUIA-Generated-Button-primary-3"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="button"
                >
                  Add
                </button>
              </div>
            </div>
          </div>
        </section>
        <section
          aria-labelledby="tabContent1"
          class="pf-c-tab-content"
          data-ouia-component-type="PF4/TabContent"
          data-ouia-safe="true"
          data-testid="git-domains-tab"
          hidden=""
          id="tabContent1"
          role="tabpanel"
          tabindex="0"
        >
          <div
            class="pf-c-tab-content__body"
          >
            <div
              class="pf-c-simple-list"
              data-testid="git-domains-list"
            >
              <ul>
                <li
                  class=""
                >
                  <a
                    class="pf-c-simple-list__item-link"
                    href=""
                    tabindex="0"
                  >
                    <div
                      class="pf-l-split"
                    >
                      <div
                        class="pf-l-split__item"
                      >
                        <span
                          class="pf-c-truncate"
                        >
                          <span
                            class="pf-c-truncate__start"
                          >
                            https://github.example-1.com
                          </span>
                        </span>
                      </div>
                      <div
                        class="pf-l-split__item pf-m-fill"
                      />
                      <div
                        class="pf-l-split__item"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 448 512"
                          width="1em"
                        >
                          <path
                            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                          />
                        </svg>
                      </div>
                    </div>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    class="pf-c-simple-list__item-link"
                    href=""
                    tabindex="0"
                  >
                    <div
                      class="pf-l-split"
                    >
                      <div
                        class="pf-l-split__item"
                      >
                        <span
                          class="pf-c-truncate"
                        >
                          <span
                            class="pf-c-truncate__start"
                          >
                            https://github.example-2.com
                          </span>
                        </span>
                      </div>
                      <div
                        class="pf-l-split__item pf-m-fill"
                      />
                      <div
                        class="pf-l-split__item"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 448 512"
                          width="1em"
                        >
                          <path
                            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                          />
                        </svg>
                      </div>
                    </div>
                  </a>
                </li>
                <li
                  class=""
                >
                  <a
                    class="pf-c-simple-list__item-link"
                    href=""
                    tabindex="0"
                  >
                    <div
                      class="pf-l-split"
                    >
                      <div
                        class="pf-l-split__item"
                      >
                        <span
                          class="pf-c-truncate"
                        >
                          <span
                            class="pf-c-truncate__start"
                          >
                            https://github.example-3.com
                          </span>
                        </span>
                      </div>
                      <div
                        class="pf-l-split__item pf-m-fill"
                      />
                      <div
                        class="pf-l-split__item"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 448 512"
                          width="1em"
                        >
                          <path
                            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                          />
                        </svg>
                      </div>
                    </div>
                  </a>
                </li>
              </ul>
            </div>
            <hr
              class="pf-c-divider pf-u-mt-md pf-u-mb-md"
            />
            <div
              class="pf-l-split"
            >
              <div
                class="pf-l-split__item form-text-input"
              >
                <form
                  class="pf-c-form"
                  novalidate=""
                >
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid="false"
                        aria-label="add github enterprise domain"
                        class="pf-c-form-control"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-4"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe="true"
                        placeholder="Add GitHub Enterprise domain"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </form>
              </div>
              <div
                class="pf-l-split__item pf-m-fill form-fill"
              />
              <div
                class="pf-l-split__item"
              >
                <button
                  aria-disabled="true"
                  aria-label="add github enterprise domain"
                  class="pf-c-button pf-m-primary pf-m-disabled"
                  data-ouia-component-id="OUIA-Generated-Button-primary-4"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="button"
                >
                  Add
                </button>
              </div>
            </div>
          </div>
        </section>
      </div>
    </section>
  </article>
</DocumentFragment>
`;

exports[`renders correctly with one domain 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <section
      class="pf-c-page__main-tabs pf-m-light pf-m-limit-width"
    >
      <div
        class="pf-c-page__main-body"
      >
        <div
          class="pf-c-tabs pf-m-page-insets"
          data-ouia-component-id="OUIA-Generated-Tabs-1"
          data-ouia-component-type="PF4/Tabs"
          data-ouia-safe="true"
          id="open-tabs-example-tabs-list"
        >
          <ul
            class="pf-c-tabs__list"
            role="tablist"
          >
            <li
              class="pf-c-tabs__item pf-m-current"
              role="presentation"
            >
              <button
                aria-controls="tabContent0"
                aria-selected="true"
                class="pf-c-tabs__link"
                data-ouia-component-type="PF4/TabButton"
                data-ouia-safe="true"
                id="pf-tab-0-open-tabs-example-tabs-list"
                role="tab"
                type="button"
              >
                <span
                  class="pf-c-tabs__item-text"
                >
                  Dev Spaces endpoints
                </span>
              </button>
            </li>
            <li
              class="pf-c-tabs__item"
              role="presentation"
            >
              <button
                aria-controls="tabContent1"
                aria-selected="false"
                class="pf-c-tabs__link"
                data-ouia-component-type="PF4/TabButton"
                data-ouia-safe="true"
                id="pf-tab-1-open-tabs-example-tabs-list"
                role="tab"
                type="button"
              >
                <span
                  class="pf-c-tabs__item-text"
                >
                  GitHub Enterprise domains
                </span>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </section>
    <section
      class="pf-c-page__main-section pf-m-light pf-m-limit-width"
    >
      <div
        class="pf-c-page__main-body"
      >
        <section
          aria-labelledby="tabContent0"
          class="pf-c-tab-content"
          data-ouia-component-type="PF4/TabContent"
          data-ouia-safe="true"
          data-testid="dev-spaces-endpoints-tab"
          id="tabContent0"
          role="tabpanel"
          tabindex="0"
        >
          <div
            class="pf-c-tab-content__body"
          >
            No endpoints added yet
            <hr
              class="pf-c-divider pf-u-mt-md pf-u-mb-md"
            />
            <div
              class="pf-l-split"
            >
              <div
                class="pf-l-split__item form-text-input"
              >
                <form
                  class="pf-c-form"
                  novalidate=""
                >
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid="false"
                        aria-label="new endpoint"
                        class="pf-c-form-control"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe="true"
                        placeholder="Add endpoint"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </form>
              </div>
              <div
                class="pf-l-split__item pf-m-fill form-fill"
              />
              <div
                class="pf-l-split__item"
              >
                <button
                  aria-disabled="true"
                  aria-label="add endpoint"
                  class="pf-c-button pf-m-primary pf-m-disabled"
                  data-ouia-component-id="OUIA-Generated-Button-primary-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="button"
                >
                  Add
                </button>
              </div>
            </div>
          </div>
        </section>
        <section
          aria-labelledby="tabContent1"
          class="pf-c-tab-content"
          data-ouia-component-type="PF4/TabContent"
          data-ouia-safe="true"
          data-testid="git-domains-tab"
          hidden=""
          id="tabContent1"
          role="tabpanel"
          tabindex="0"
        >
          <div
            class="pf-c-tab-content__body"
          >
            <div
              class="pf-c-simple-list"
              data-testid="git-domains-list"
            >
              <ul>
                <li
                  class=""
                >
                  <a
                    class="pf-c-simple-list__item-link"
                    href=""
                    tabindex="0"
                  >
                    <div
                      class="pf-l-split"
                    >
                      <div
                        class="pf-l-split__item"
                      >
                        <span
                          class="pf-c-truncate"
                        >
                          <span
                            class="pf-c-truncate__start"
                          >
                            https://github.example.com
                          </span>
                        </span>
                      </div>
                      <div
                        class="pf-l-split__item pf-m-fill"
                      />
                      <div
                        class="pf-l-split__item"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style="vertical-align: -0.125em;"
                          viewBox="0 0 448 512"
                          width="1em"
                        >
                          <path
                            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                          />
                        </svg>
                      </div>
                    </div>
                  </a>
                </li>
              </ul>
            </div>
            <hr
              class="pf-c-divider pf-u-mt-md pf-u-mb-md"
            />
            <div
              class="pf-l-split"
            >
              <div
                class="pf-l-split__item form-text-input"
              >
                <form
                  class="pf-c-form"
                  novalidate=""
                >
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid="false"
                        aria-label="add github enterprise domain"
                        class="pf-c-form-control"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-2"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe="true"
                        placeholder="Add GitHub Enterprise domain"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </form>
              </div>
              <div
                class="pf-l-split__item pf-m-fill form-fill"
              />
              <div
                class="pf-l-split__item"
              >
                <button
                  aria-disabled="true"
                  aria-label="add github enterprise domain"
                  class="pf-c-button pf-m-primary pf-m-disabled"
                  data-ouia-component-id="OUIA-Generated-Button-primary-2"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  disabled=""
                  type="button"
                >
                  Add
                </button>
              </div>
            </div>
          </div>
        </section>
      </div>
    </section>
  </article>
</DocumentFragment>
`;
